src/Syllable.php

```diff
@@ -754,13 +754,28 @@ private function parseWord($word)
             return [$word];
         }
 
+        $wordLowerCased = mb_strtolower($word);
+
         // Is it a pre-hyphenated word?
-        if (isset($this->hyphenation[$word])) {
-            return mb_split('-', $this->hyphenation[$word]);
+        if (isset($this->hyphenation[$wordLowerCased])) {
+            $parts = [];
+            $part = '';
+            for ($i = 0, $j = 0; $i < mb_strlen($this->hyphenation[$wordLowerCased]); $i++) {
+                if (mb_substr($this->hyphenation[$wordLowerCased], $i, 1) !== '-') {
+                    $part .= mb_substr($word, $j++, 1);
+                } else {
+                    $parts[] = $part;
+                    $part = '';
+                }
+            }
+            if (!empty($part)) {
```

Comment:
`if ($part !== '')`

Reply:
It orients itself to similar code places. See for example the end of parseWordByPatterns(). Would adjust it for all places or leave it as is.